### PR TITLE
fix empty ips in alt hosts of agent /ndsz output

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/istio-agent/agent.go [hunk 1]
```diff
@@ -546,10 +546,14 @@ func (a *Agent) GetDNSTable() *dnsProto.NameTable {
 				if _, exists := nt.Table[host]; !exists {
 					addresses := make([]string, len(ipv4)+len(ipv6))
 					for _, addr := range ipv4 {
-						addresses = append(addresses, addr.String())
+						if len(addr.String()) > 0 {
```
Comment (hzxuzhonghu) [inline pkg/istio-agent/agent.go:549]: `ipv4` slice item here should not be empty

Comment (hzxuzhonghu) [reply pkg/istio-agent/agent.go:549]: @zhlsunshine

Comment (ramaraochavali) [reply pkg/istio-agent/agent.go:549]:
```
        "a-benjamin-sandbox-test-ingress.armada-a-benjamin-sandbox-test.": {
            "ips": [
                "",
                "172.20.216.52"
            ],
            "registry": "Kubernetes"
        },
        "a-benjamin-sandbox-test-ingress.armada-a-benjamin-sandbox-test.svc.": {
            "ips": [
                "",
                "172.20.216.52"
            ],
            "registry": "Kubernetes"
        },
```
It happens for alt hosts in. Let me debug this a bit more

Comment (ramaraochavali) [reply pkg/istio-agent/agent.go:549]: https://github.com/istio/istio/blob/1607f91e1c5c85b6927bebca591b9aafc4b16e05/pkg/dns/client/dns.go#L600 - We handle this case while building actual DNS responses. But here it is missed.

Comment (ramaraochavali) [reply pkg/istio-agent/agent.go]: Ok. I debugged this more, it is not with empty IPs but with len vs. capacity :-)

Comment (zhlsunshine) [reply pkg/istio-agent/agent.go]: > Ok. I debugged this more, it is not with empty IPs but with len vs. capacity :-)

Hi @hzxuzhonghu & @hzxuzhonghu, does it mean that there may be some invalidation IP entries?

Comment (ramaraochavali) [reply pkg/istio-agent/agent.go]: It is only in debug endpoint, not on the actual code

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — COMMENTED** [review]
lgtm

**Pull request merged**